packages/realm/src/Object.ts

```diff
@@ -120,13 +122,52 @@ class RealmObject<T = DefaultObject> {
   entries(): [string, unknown][] {
     throw new Error("Not yet implemented");
   }
-  toJSON(): unknown {
-    // return { ...this };
+
+  toJSON(_?: string, cache: JSONCacheMap = new Map()): DefaultObject {
+    // Construct a reference-id of table-name & primaryKey if it exists, or fall back to objectId.
+
+    // Check if current objectId has already processed, to keep object references the same.
+    const existing = this._getOwnCache(cache);
+    if (existing) {
+      return existing;
+    }
+
+    // Create new result, and store in cache.
+    const result: DefaultObject = {};
+    this._setOwnCache(cache, result);
+
+    // Move all enumerable keys to result, triggering any specific toJSON implementation in the process.
+    // TODO: Consider Object.keys(this)
+    //   .concat(Object.keys(Object.getPrototypeOf(this)))
+    //   .forEach((key) => {
     for (const key in this) {
       const value = this[key];
-      console.log({ key, value });
+
+      // skip any functions & constructors (in case of class models).
+      if (typeof value === "function") {
+        continue; // continue
+      }
+
+      if (value === null || value === undefined) {
+        result[key] = value;
+      } else if (value instanceof RealmObject || value instanceof Realm.Collection) {
+        // recursively trigger `toJSON` for Realm instances with the same cache.
+        result[key] = value.toJSON(key, cache);
+      } else if (value instanceof Realm.Dictionary /* TODO: value[Symbol.IS_PROXIED_DICTIONARY] */) {
+        // TODO: Consider dictionary compatibility comment.
```

Comment:
I am quite confused by this `Symbol.IS_PROXIED_DICTIONARY` comment / information. there also were no dictionary tests so I think I would need to add them once they are implemented.